xbmc/addons/kodi-addon-dev-kit/include/kodi/libXBMC_addon.h

```diff
@@ -155,6 +155,7 @@ typedef struct CB_AddOn
   int (*GetFileChunkSize)(const void* addonData, void* file);
   bool (*FileExists)(const void* addonData, const char *strFileName, bool bUseCache);
   int (*StatFile)(const void* addonData, const char *strFileName, struct __stat64* buffer);
+  char *(*GetFileProperty)(const void* addonData, void* file, XFILE::FileProperty type, const char *name);
```

Comment:
The "type" parameter introduces a non Kodi devkit internal C++ type to the public C API , which is a nogo. It even has a namespace, which is a C++ language construct. Unfortunately, we already do this at other places in this file, but I think we can do better. Correct solution is to introduce a new enum for the type in this header file.

But as this API is deprecated  and supposed to die "soon", it's up to you to change this or not. Just saying.

Reply:
I was a little bit surprised that we use XFILE things here, looking at the code IFileTypes.h is copied into the addon dev kit during binary addon installation step.

Sure, nice is different. but I thougth as well that this interface is a question of month befor it dies.

Reply:
Yeah it's a mess and good  that it will (hopefully) die soon.